answers.md

```diff
@@ -1,19 +1,21 @@
 ### All Answers to Partner Study should be filled out in this file.
 
 1. Single Page Application
-
+    A way to design a web page where you do not need to refresh the browser to see DOM changes.
 2. Compilers
-
+    A program that interprets human readable code to browser or machine readable code or binary.
 3. Bundlers
-
+    Bundlers are programs that re-organize your code files into more efficient or more conventional ways.
 4. Elements
-
+    An element inside of React is typically a piece of html(?) that is returned from components.
 5. Components
-
+    Components are basically classes inside of React. Re-useable pieces of code.
 6. JSX
-
+    A library of sorts that adds functionality to Javascript, such as the ability to define variables as HTML elements.
 7. Package Mangers
-
+    A tool that allows you to keep track of different packages on a project. For example, NPM manages all of the different node libraries you may have on a project.
 8. CDN
-
+    Content delivery network, a way to access files without downloading them locally, through the internet.
 9. Props and State
```

Comment:
It's very important you understand what state and props are and what their differences are.

Similarities:
— state and props are both plain JS objects
— state and props are just data being passed to components

Differences:
— state is stored in the class Component
— props are passed to child components (either functional/class components can receive props)
— state is mutable but only inside the class Component it's in
— props immutable

There's more to it. Please read up more on state and props [1](https://reactjs.org/docs/components-and-props.html) [2](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)